samples/ck-rx65n/basic-sample/src/iotconnect_app.c

```diff
@@ -175,15 +241,19 @@ bool app_startup(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_ptr) {
 	dp.cert_size = sample_device_cert_len;
 	dp.key = (uint8_t *) (sample_device_private_key_ptr);
 	dp.key_size = sample_device_private_key_len;
-	dp.crypto_method = &crypto_method_ec_secp256;
+#ifdef ECC_CRYPTO

```

Comment:
What do you think about doing a runtime switch with driver parameters? Basically, the application code decides  at runtime (with driver parameters) which certificate type to use. You could find similar mechanism in TO driver clot switching.

Reply:
In this case we would need to parse the X509 cert so that we could set the crypto_method dynamically at runtime in the driver parameters (of the s/w x509 auth driver in this case). The s/w auth driver could even parse the certificate so that there is no need for the caller to populate crypto_method. Do you think it would be better to do that in another ticket, as the s/w auth driver is common to all platforms?